File: app/src/main/java/d/d/meshenger/Log.java
package d.d.meshenger;

/*
 * Wrapper for android.util.Log to disable logging
 */
public class Log {
    private static String contextString(Object context) {
        if (context instanceof String) {
            return (String) context;
        } else {
            return context.getClass().getSimpleName();
        }
    }

    public static void d(Object context, String message) {
        if (BuildConfig.DEBUG) {
            String tag = contextString(context);
            android.util.Log.d(tag, message);
        }
    }
}

File: app/src/main/java/d/d/meshenger/Log.kt
package d.d.meshenger

import android.util.Log

/*
* Wrapper for android.util.Log to disable logging
*/
object Log {
    private fun contextString(context: Any): String {
        if (context is String) {
            return context
        } else {
            return context.javaClass.simpleName
        }
    }

    fun d(context: Any, message: String) {
        if (BuildConfig.DEBUG) {
            val tag = contextString(context)
            Log.d(tag, message)
        }
    }

    fun i(context: Any, message: String) {
        if (BuildConfig.DEBUG) {
            val tag = contextString(context)
            Log.i(tag, message)
        }
    }

    fun w(context: Any, message: String) {
        if (BuildConfig.DEBUG) {
            val tag = contextString(context)
            Log.w(tag, message)
        }
    }

    fun e(context: Any, message: String) {
        val tag = contextString(context)
        Log.e(tag, message)
    }
}
